### ISSUE TITLE
V13.2.3 - remove "referer request header check" as defence method

### ISSUE THREAD

**Issue by elarlang** [OWASP/ASVS#475]
**requirement**
13.2.3 Verify that RESTful web services are protected from Cross-Site Request Forgery via the use of at least one or more of the following: double submit cookie pattern, CSRF nonces, ORIGIN request header checks, and referer request header checks.

**problem**
User-agents (ex Firefox) nowadays may give users possibility to disable sending user-agent header.

It makes sense to user referer request header for monitoring purposes - example: if there is post request from other domain, it's anyway suspicious. But if there is no referer header in request, then it's a bit risky to consider it automatically as dangerous request.

**recommendation**
Remove referer request header check.

**Comment by vanderaj** [comment]
User agents are different than the referer header though. Referer is controlled by the "Referrer-Policy" header, or by the browser if it decides to put it in. Does this change your view on this issue?

**Comment by elarlang** [comment]
Yes, I know the difference and by "user-agent" here I meant browser (not user-agent header). So, it does not change my view.

From some of my sites I have used referrer checks and some users/browsers have configured to not send it at all or send only same-origin. It means, defence can not relay on referer header - it may cause availability risks for such users.

First response from Google: https://www.technipages.com/firefox-enable-disable-referrer

Additionally - we just added new verification request to ASVS4 also with recommendation to use "no-referrer". From that perspective it does not make sense to recommend to use referer header in other verification request.
